flux-middle/src/global_env.rs

```diff
@@ -27,6 +27,7 @@ pub struct GlobalEnv<'genv, 'tcx> {
     qualifiers: Vec<rty::Qualifier>,
     uifs: FxHashMap<Symbol, rty::UifDef>,
     fn_sigs: RefCell<FxHashMap<DefId, rty::PolySig>>,
+    fn_quals: FxHashMap<DefId, Vec<SurfaceIdent>>,
```

Comment:
Maybe 
```suggestion
    fn_quals: FxHashMap<DefId, FxHashSet<String>>,
```
Also, perhaps a  doc comment saying this is the set of qualifiers for a function.